sklearn/manifold/_t_sne.py

```diff
@@ -564,7 +564,8 @@ class TSNE(BaseEstimator):
         is used in other manifold learning algorithms. Larger datasets
         usually require a larger perplexity. Consider selecting a value
         between 5 and 50. Different values can result in significantly
-        different results.
+        different results. It is important to never use a value larger than
+        the number of samples in your data.
```

Comment:
Nit: I prefer to not say something is important without describing why. I think we can directly state the constraint:

```suggestion
        different results. The perplexity must be less that the number
        of samples.
```